MaxText/layers/linears.py

```diff
@@ -350,10 +352,10 @@ def unpermute(self, intermediate, sorted_selected_experts, weights):
 
     unsort_intermediate = jnp.take(intermediate, indices=jnp.argsort(sorted_selected_experts), axis=0)
     reshaped_weights = jnp.reshape(weights, (-1, self.num_experts_per_tok))
-    reshaped_intermediate = jnp.reshape(unsort_intermediate, (-1, self.num_experts_per_tok, self.config.emb_dim))
+    reshaped_intermediate = jnp.reshape(unsort_intermediate, (-1, self.num_experts_per_tok, self.config.emb_dim // self.config.ici_tensor_parallelism))
```

Comment:
Although we will probably never use TP over DCN, can we replace ici_tensor_parallelism with the product of ici_tensor_parallelism and dcn_tensor_parallelism here and below? (Probably best to extract into a variable first, e.g. tp=self.config.ici_tensor_parallelism * self.config.dcn_tensor_parallelism)

Reply:
Sure, will update.